src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java

```diff
@@ -365,17 +366,14 @@ private String getAddress(Principal principal) {
     public Uni<List<BehaviorGroup>> findBehaviorGroupsByBundleId(@Context SecurityContext sec, @PathParam("bundleId") UUID bundleId) {
         return getAccountId(sec)
                 .onItem().transformToUni(accountId -> behaviorGroupResources.findByBundleId(accountId, bundleId))
-                .onItem().call(behaviorGroups -> behaviorGroups.size() == 0 ? Uni.createFrom().voidItem() : Uni.combine().all().unis(
-                        behaviorGroups.stream()
+                .onItem().call(behaviorGroups -> endpointResources.loadProperties(
+                        behaviorGroups
+                                .stream()
                                 .map(BehaviorGroup::getActions)
-                                .map(bga -> bga.size() == 0 ? Uni.createFrom().voidItem() : Uni.combine().all().unis(
-                                        bga.stream()
-                                                .map(BehaviorGroupAction::getEndpoint)
-                                                .map(endpoint -> endpointResources.loadProperties(endpoint))
-                                                .collect(Collectors.toList())
-                                        )
-                                                .discardItems()
-                                ).collect(Collectors.toList())
-                ).discardItems());
+                                .flatMap(Collection::stream)
```

Comment:
I would add `.filter(Objects::nonNull)` before `.flatMap(Collection::stream)` just to be safe. There's no NPE now but the code may evolve in a way that will cause a NPE later.

Reply:
Do we have null endpoints? I wondering if adding this filter would potentially hide other issue.

Reply:
That would filter out behavior groups when `getActions() == null` and prevent a NPE throw when `.stream()` is collected on the `null` collection. We don't have that right now, but who knows in the future... :)